packages/developer-portal/src/components/pages/app-detail/pipeline/create-pipeline.tsx

```diff
@@ -0,0 +1,229 @@
+import React, { useState } from 'react'
+import { cx } from '@linaria/core'
+import {
+  BodyText,
+  Button,
+  ButtonGroup,
+  ColSplit,
+  elM6,
+  elMb10,
+  elMt10,
+  FlexContainer,
+  FormLayout,
+  Grid,
+  Input,
+  InputAddOn,
+  InputGroup,
+  InputWrap,
+  InputWrapFull,
+  Label,
+  Modal,
+  PersistantNotification,
+  Subtitle,
+  Title,
+} from '@reapit/elements'
+import { IconContainer } from '../../webhooks/__styles__'
+import { WebhooksAnimatedNewIcon } from '../../webhooks/webhooks-animated-new-icon'
+import { WebhooksAnimatedDocsIcon } from '../../webhooks/webhooks-animated-docs-icon'
+import { ExternalPages, openNewPage } from '@/utils/navigation'
+import { mixed, object, string } from 'yup'
+import { AppTypeEnum, PackageManagerEnum, PipelineModelInterface } from '@reapit/foundations-ts-definitions'
+import { httpsUrlRegex, UpdateActionNames } from '@reapit/utils-common'
+import { useForm } from 'react-hook-form'
+import { yupResolver } from '@hookform/resolvers/yup'
+import errorMessages from '@/constants/error-messages'
+import { useReapitUpdate } from '@reapit/utils-react'
+import { reapitConnectBrowserSession } from '../../../../core/connect-session'
+import { useReapitConnect } from '@reapit/connect-session'
+
+export const pipelineCreateFormHandle =
+  (createPipeline: (values: Partial<PipelineModelInterface>) => Promise<boolean>, refresh: () => void) =>
+  async (values: PipelineModelInterface) => {
+    const result = await createPipeline({
+      ...values,
+      appType: AppTypeEnum.REACT,
+      name: values.repository,
+    })
+
+    if (result) refresh()
+  }
+
+const PipelineCreationModal = ({
```

Comment:
I would probably move to it's own component

Reply:
And defining an interface for the props is a bit more readable

Reply:
(y)